### PR TITLE
cleanup and polish codebase, improve test coverage and documentation

### PR DESCRIPTION
**Pull request by j3t**
- add tests for public api methods
- remove OkHttp as dependency from Cloud and Template
- fix that ACL token is not sent if necessary
- fix imports

This PR prepares upcoming changes (e.g. #107) and makes it easier to maintain the codebase.

### CODE REVIEW

**Review by multani — APPROVED** [review]
Thanks for the cleanup!

File: src/main/java/org/jenkinsci/plugins/nomad/NomadApi.java [hunk 1]
```diff
@@ -29,42 +36,191 @@
...
+     * Validates a given {@link NomadWorkerTemplate}.
+     * @return FormValidation object with kind = OK or ERROR and a message.
+     */
+    public FormValidation validateTemplate(NomadWorkerTemplate template) {
+        String id = UUID.randomUUID().toString();
+
+        Request request = createRequestBuilder()
+                .url(cloud.getNomadUrl() + "/v1/job/"+id+"/plan")
+                .post(RequestBody.create(buildWorkerJob(id, "", template), JSON))
+                .build();
```
[the remark this answers is not in the document]
Comment (j3t) [reply src/main/java/org/jenkinsci/plugins/nomad/NomadApi.java:74]: Yes, the output of the /plan endpoint is much more descriptive than from the /validate endpoint (see https://github.com/jenkinsci/nomad-plugin/pull/82#issuecomment-949560763).

File: src/main/java/org/jenkinsci/plugins/nomad/NomadApi.java [hunk 2]
```diff
@@ -29,42 +36,191 @@
...
 
-    JobInfo[] getJobs(Request request) {
-        JobInfo[] jobs;
+    /**
+     * Checks whether Nomad is reachable.
+     * @return FormValidation object with kind = OK or ERROR and a message.
+     */
+    public FormValidation checkConnection() {
+        Request request = createRequestBuilder()
+                .url(cloud.getNomadUrl() + "/v1/agent/self")
```
[the remark this answers is not in the document]
Comment (j3t) [reply src/main/java/org/jenkinsci/plugins/nomad/NomadApi.java:45]: Good point! I will change this.

File: src/main/java/org/jenkinsci/plugins/nomad/NomadApi.java [hunk 3]
```diff
@@ -29,42 +36,188 @@
...
+    }
+
+    /**
+     * Validates a given {@link NomadWorkerTemplate}.
+     * @return FormValidation object with kind = OK or ERROR and a message.
+     */
+    public FormValidation validateTemplate(NomadWorkerTemplate template) {
+        String id = UUID.randomUUID().toString();
+
+        Request request = createRequestBuilder("/v1/job/"+id+"/plan")
```
Comment (multani) [inline src/main/java/org/jenkinsci/plugins/nomad/NomadApi.java:70]:
```suggestion
        Request request = createRequestBuilder("/v1/job/" + id + "/plan")
```

**Review by multani — APPROVED** [review]

**Pull request merged**